src/python/gem5/resources/resource.py

```diff
@@ -964,10 +964,22 @@ def obtain_resource(
             for key in resource_json["resources"].keys():
                 assert isinstance(key, str)
                 value = resource_json["resources"][key]
-                assert isinstance(value, str)
-                params[key] = obtain_resource(
-                    value,
-                )
+
+                if isinstance(value, str):
+                    warn(
+                        "Deprecation warning: resources field in workloads has changed"
+                        "from { category: id } to"
+                        "{ category: { id: id, resource_version: resource_version } }"
+                        "Please update your resource.json file to reflect this change."
```

Comment:
This could be printed if someone is using the database, right? How would that person update their `resources.json`?

Reply:
Yes, should I change it to "Please update the database" or "Please update the workload JSON objects"?

Reply:
You should give a warning that will be helpful to the user. If you said "Please update the database" what could the user possibly do about this? How would they interpret that message?

I don't think you should tell the user to do anything if they cannot take any actions.

*Possibly* if the user is using a local version of a database (e.g., a specialized json file), *only then* should you suggest a change, since only then they could actually take an action.

Reply:
I don't think any warning should be given. This patch is to keep this version of gem5 compatible with both versions the old and the newer version the schema. The warning should be to us that we need to do something about this and keep track of it.

I would hope the schema versioning functionality is implemented soon so we can more easily keep track of these changes in a logical way.

Reply:
The if condition change is to keep develop working with both versions of workload, I think we will remove it before we release 23.1 as when we release a new version the older workload will work with 23.0 and the new one will work with 23.1.

Reply:
> The if condition change is to keep develop working with both versions of workload, I think we will remove it before we release 23.1 as when we release a new version the older workload will work with 23.0 and the new one will work with 23.1.

I would like that but we are running low on time.